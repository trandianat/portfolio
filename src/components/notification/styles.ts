import { css } from '@emotion/react';
import { Color, Status } from 'utils/constants';
import { color } from 'utils/map';

export const notification = (variant: Status) =>
  css({
    backgroundColor: Color.WHITE,
    border: `1px solid ${color(variant)}`,
    borderLeft: `8px solid ${color(variant)}`,
    borderRadius: 6,
    bottom: 24,
    boxShadow: 'var(--shadow)',
    color: color(variant),
    font: '.875rem var(--sans)',
    left: 24,
    maxWidth: 'calc(80% - 12px)',
    padding: '12px 16px',
    position: 'fixed',
    wordWrap: 'break-word',
  });
