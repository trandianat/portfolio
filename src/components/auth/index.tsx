import { useAuthenticator, withAuthenticator } from '@aws-amplify/ui-react';
import { Fragment, ReactNode } from 'react';
import { Link, useLocation } from 'react-router-dom';
import * as styles from 'components/auth/styles';
import isProjects from 'utils/projects';

const Auth = ({ children }: { children: ReactNode }): JSX.Element => {
  const { signOut } = useAuthenticator();
  const { pathname } = useLocation();
  return (
    <Fragment>
      <div css={styles.navigation}>
        {isProjects(pathname) ? <div /> : <Link to="projects">â‰ª Back</Link>}
        <button onClick={signOut}>Sign out</button>
      </div>
      {children}
    </Fragment>
  );
};

export default withAuthenticator(Auth, {
  components: { SignIn: { Footer: () => <Fragment /> } },
  formFields: {
    signIn: {
      username: { label: '', placeholder: 'Username' },
      password: { label: '', placeholder: 'Password' },
    },
  },
  hideSignUp: true,
  // services: {
  //   handleSignIn: async ({
  //     username,
  //     password,
  //   }: {
  //     username: string;
  //     password: string;
  //   }) => console.log('Credentials: ' + username + ' / ' + password),
  // },
});
