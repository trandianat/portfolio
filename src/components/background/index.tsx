import { PropsWithChildren } from 'react';
import * as styles from 'components/background/styles';
import { Color } from 'utils/types';

export const Background = ({
  children,
  color = Color.WHITE,
}: PropsWithChildren<{ color?: Color }>): JSX.Element => {
  return (
    <div css={styles.background(color)}>
      <div css={styles.container}>{children}</div>
    </div>
  );
};
