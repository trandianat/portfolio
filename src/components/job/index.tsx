import * as styles from 'components/job/styles';

export const Job = ({
  role,
  company,
  start,
  end,
  skills,
  descriptions,
  indent = false,
}: {
  role: string;
  company: string;
  start: string;
  end: string;
  skills: string[];
  descriptions: string[];
  indent?: boolean;
}): JSX.Element => (
  <div css={styles.job(indent)}>
    <div css={styles.dates}>
      <div css={styles.range}>
        <div className="sans">{end}</div>
        <div className="sans">{start}</div>
      </div>
      <div css={styles.timeline}>
        <div css={styles.circle} />
        <div css={styles.line} />
        <div css={styles.circle} />
      </div>
    </div>
    <div css={styles.role}>
      <h2>{role}</h2>
      <p className="sans">{company}</p>
      <div css={styles.skills}>
        {skills.map((skill: string) => (
          <div css={styles.skill} key={skill}>
            {skill}
          </div>
        ))}
      </div>
      <ul>
        {descriptions.map((description: string) => (
          <li key={description}>{description}</li>
        ))}
      </ul>
    </div>
  </div>
);
