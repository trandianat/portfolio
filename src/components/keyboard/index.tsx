import { useEffect, useState } from 'react';
import * as styles from 'components/keyboard/styles';
import { Keyboard as Key } from 'utils/constants';

export const Keyboard = ({
  disabled,
  guesses,
  handleSubmit,
  input,
  setInput,
}: {
  disabled: boolean;
  guesses: string[];
  handleSubmit: any;
  input: string;
  setInput: any;
}): JSX.Element => {
  const [rows, setRows] = useState<Array<string[]>>([]);

  const isAction = (key: string): boolean =>
    [Key.DELETE, Key.ENTER].includes(key as Key);
  const isDelete = (key: string): boolean => key === Key.DELETE;
  const isEnter = (key: string): boolean => key === Key.ENTER;

  const setLetterClass = (key: string, rowIndex: number): string => {
    let className = 'letter';
    if (isAction(key)) {
      className += ' action';
      if (rowIndex < rows.length - 1) {
        className += ' primary';
      }
    } else {
      const used = guesses.map(guess => [...guess]).flat();
      if (used.includes(key)) {
        className += ' used';
      }
    }
    return className;
  };

  useEffect(() => {
    const row1 = ['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'];
    const row2 = ['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'];
    const row3 = [Key.ENTER, 'Z', 'X', 'C', 'V', 'B', 'N', 'M', Key.DELETE];
    const row4 = [Key.ENTER, Key.DELETE];
    setRows([row1, row2, row3, row4]);
  }, []);

  return (
    <div css={styles.keyboard}>
      {rows.map((row, index) => (
        <div className={`row row-${index + 1}`} key={`row-${index + 1}`}>
          {row.map((key: string) => (
            <button
              className={setLetterClass(key, index)}
              key={key}
              onClick={() => {
                if (isDelete(key)) {
                  setInput((input: string) => input.slice(0, -1));
                } else if (isEnter(key)) {
                  handleSubmit();
                } else {
                  setInput((input: string) => input + key);
                }
              }}
              {...(((isDelete(key) && !input.length) ||
                (isEnter(key) && input.length < 5) ||
                (!isAction(key) && input.length === 5) ||
                disabled) && {
                disabled: true,
              })}
            >
              {key}
            </button>
          ))}
        </div>
      ))}
    </div>
  );
};
