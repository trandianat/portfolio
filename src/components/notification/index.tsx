import { PropsWithChildren } from 'react';
import * as styles from 'components/notification/styles';
import { Status } from 'utils/constants';

export const Notification = ({
  children,
  variant = Status.DEFAULT,
}: PropsWithChildren<{ variant?: Status }>): JSX.Element => {
  return <div css={styles.notification(variant)}>{children}</div>;
};
