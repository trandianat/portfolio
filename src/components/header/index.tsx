import { ReactNode } from 'react';
import { Background } from 'components/background';
import * as styles from 'components/header/styles';

export const Header = ({ children }: { children: ReactNode }): JSX.Element => (
  <Background>
    <div css={styles.header}>
      <div css={styles.divider} />
      <div css={styles.name}>
        <div>
          <h1>Diana Tran</h1>
          <h2 className="sans">Software Engineer & Manager</h2>
        </div>
        <div css={styles.logo}>
          <div className="content">
            <div className="dot" />
            <div>DT</div>
            <div className="line" />
          </div>
        </div>
      </div>
      {children}
    </div>
  </Background>
);
