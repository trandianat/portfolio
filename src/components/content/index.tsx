import { Fragment, ReactNode } from 'react';
import { Background } from 'components/background';
import { Header } from 'components/header';
import { Navigation } from 'components/navigation';
import { Color } from 'utils/constants';

export const Content = ({ children }: { children: ReactNode }): JSX.Element => (
  <Fragment>
    <Header />
    <Background color={Color.GRAY}>
      <Navigation />
      {children}
    </Background>
  </Fragment>
);
