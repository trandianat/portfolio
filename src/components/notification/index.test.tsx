import { create } from 'react-test-renderer';
import { Notification } from 'components/notification';

describe('Navigation', () => {
  it('should match the snapshot', () => {
    expect(create(<Notification>Message</Notification>)).toMatchSnapshot();
  });
});
