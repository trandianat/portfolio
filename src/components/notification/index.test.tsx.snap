// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
.emotion-0 {
  background-color: var(--white);
  border: 1px solid var(--black);
  border-left: 8px solid var(--black);
  border-radius: 6px;
  bottom: 24px;
  box-shadow: var(--shadow);
  color: var(--black);
  font: .875rem var(--sans);
  left: 24px;
  max-width: calc(80% - 12px);
  padding: 12px 16px;
  position: fixed;
  word-wrap: break-word;
}

<div
  className="emotion-0"
>
  Message
</div>
`;
